mailpoet/lib/Cron/CronTrigger.php

```diff
@@ -42,10 +42,14 @@ public function __construct(
   public function init() {
     $currentMethod = $this->settings->get(self::SETTING_NAME . '.method');
     try {
+      if ($currentMethod === self::METHOD_ACTION_SCHEDULER) {
+        $this->cronActionScheduler->init();
+        return true;
+      } else {
+        $this->cronActionScheduler->deactivate();
```

Comment:
We could also alternatively do the deactivation in the job run (like "IF current method is not AS, deactivate"). Would that be better/worse or similar?

Reply:
Moving it to the action itself would require always registering the action even if the method is not active. On the other hand running deactivate on every requests seems to trigger an SQL query. So I'm going to add `deactivateOnTrigger` method that will register the trigger action but with different callback, which only unschedules.

Reply:
> Moving it to the action itself would require always registering the action even if the method is not active.

Related to my [question](https://github.com/mailpoet/mailpoet/pull/4159#discussion_r930072008) — why would it always require registering the action? It seems to me that:
1. If `DaemonTrigger::NAME` is registered in Action Scheduler, and therefore running, we would detect and deactivate all scheduled Action Scheduler jobs in its `action_scheduler_after_process_queue`.
2. `DaemonTrigger::NAME` is **not** registered in Action Scheduler, then it has been already unregistered and we don't need to try to unregister it at all. Or is that not right? 🤔

Reply:
There are two things you need to register.
1) action in action scheduler
2) WordPress callback for the action
Currently both happens in the `$this->cronActionScheduler->init();` I used wrong wording by "registering the action" I meant the WP action (aka hook), not the scheduler action. Sorry for the confusion.

So for the case, the action is registered in the action scheduler, but the method is not active anymore in plugin settings I still need to register some WP action that will handle the deactivation. I can move the whole logic to `$this->cronActionScheduler->init();` and call it on every request, or the other place I found reasonable is the cron trigger as it handles logic around activating different methods.
In the next steps, we will remove the visitor method. So with that, we could get rid of the whole cron trigger and use only `$cronActionScheduler->init();`.
Does that make sense?

Reply:
@costasovo Ahh, I get it — the scheduled action can't unregister itself because the code to handle that scheduled action is no longer registered. Makes sense.